### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by sigiesec**
See individual commits.

### CODE REVIEW

File: com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend [hunk 1]
```diff
@@ -175,9 +173,107 @@ class CppGeneratorTest extends AbstractGeneratorTest
...
+    }
+
+    @Test
+    def void testBasicDispatcherCmake()
+    {
+        val defaultGenerationSettingsProvider = generationSettingsProvider as DefaultGenerationSettingsProvider
+        defaultGenerationSettingsProvider.reset() // TODO remove this, it is necessary because the dependencies are reused across test cases
+        defaultGenerationSettingsProvider.projectSetFactory = new CMakeProjectSetFactory
+        defaultGenerationSettingsProvider.moduleStructureStrategy = new CABModuleStructureStrategy
+        val fileCount = 17
```
Comment (schregger) [inline com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend]: A comment would be nice

File: com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend [hunk 2]
```diff
@@ -175,9 +173,107 @@ class CppGeneratorTest extends AbstractGeneratorTest
...
+                name = "Test"
+                version= version_name("0.1.0-unreleased")
+                url = "TODO"
+                description = """
+                TODO
+                """
+                
+                build_requires = "CMakeMacros/0.3.latest@cab/testing"
+                requires = ( 
+                            ("BTC.CAB.Commons/1.9.latest@cab/testing"),
```
Comment (schregger) [inline com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend]: Shouldn't this come from a config file or similar?

Comment (sigiesec) [reply com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend]: What should come from a config file? This is the expected test outcome. Are you referring to the generator? 

The BTC.CAB.Commons version is determined by the BTC.CAB.ServiceComm version that is used, and that is part of the generation settings.

File: com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend (hunk 1 above)
Comment (sigiesec) [reply com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/cpp/CppGeneratorTest.xtend]: On what? How the file count is determined? This is a general issue we already discussed before. In this regard, it is a regression test, which checks if the file count stays the same over the run where this was originally determined.

**Comment by sigiesec** [comment]
@huttenlocher Although this is now merged, maybe you can give the changes a look next week?

**Pull request merged**